cognite/client/data_classes/capabilities.py

```diff
@@ -245,15 +254,23 @@ def _infer_project(self, project: str | None = None) -> str:
 
     def as_tuples(self, project: str | None = None) -> set[tuple]:
         project = self._infer_project(project)
-        return set().union(
-            *(
-                proj_cap.capability.as_tuples()
-                for proj_cap in self
-                if isinstance(proj_cap.project_scope, AllProjectsScope)
+
+        output: set[tuple] = set()
+        for proj_cap in self:
+            cap = proj_cap.capability
+            if isinstance(cap, UnknownAcl):
+                warnings.warn(f"Unknown capability {cap.capability_name} will be ignored in comparison")
+                continue
+            if isinstance(cap, LegacyCapability):
+                # Legacy capabilities should be encouraged to be migrated to the new format, so we skip them.
```

Comment:
Just a suggestion (+ same above)
```suggestion
                # Legacy capabilities are no longer in use so they are safe to skip
```